app/charts/area/chart-area.tsx

```diff
@@ -41,44 +37,59 @@ export const ChartAreasVisualization = ({
   queryFilters: QueryFilters;
 }) => {
   const locale = useLocale();
-  const [{ data, fetching, error }] = useDataCubeObservationsQuery({
-    variables: {
-      locale,
-      iri: dataSetIri,
-      dimensions: null, // FIXME: Try to load less dimensions
-      filters: queryFilters,
-    },
-  });
+  const { setChartError } = useChartError();
+  const [{ data, fetching, error: dataLoadingError }] =
+    useDataCubeObservationsQuery({
+      variables: {
+        locale,
+        iri: dataSetIri,
+        dimensions: null, // FIXME: Try to load less dimensions
+        filters: queryFilters,
+      },
+    });
 
   const observations = data?.dataCubeByIri?.observations.data;
+  const observationsPresent = useMemo(
+    () =>
+      observations
+        ? observations.length > 0 && observations.map((d) => d.y).some(isNumber)
+        : false,
+    [observations]
+  );
```

Comment:
I think it is a good as a first step.

For future improvement: I wonder if the data cubes observations query should not be handled by the hook directly so the whole data fetching / checking is done in a single hook, so that every chart can then use it easily.

const { data, error } = useChartData({
  dimensions,
  iri,
filters, 
  checkDatum: (d) => isNumber(d)
})

Reply:
Awesome idea, I have made an initial implementation 👌

Reply:
Btw. `observations.map((d) => d.y).some(isNumber)` seems long outdated, as `Observation` is` [dimensionIri: dimensionValue]`, there is no arbitrary "x" or "y" fields.